tests/helpers/generate-table.js

```diff
@@ -10,7 +11,12 @@ import {
 // reexport for use in tests
 export { configureTableGeneration, resetTableGenerationConfig, generateColumns, generateRows };
 
-const fullTable = hbs`
+function actionString(actionName) {
+  let quotedActionName = `"${actionName}"`;
+  return SUPPORTS_CLOSURE_ACTIONS ? `(action ${quotedActionName})` : quotedActionName;
```

Comment:
Oh dang, this is clever but it is too much.

We need to test both versions of action explicitly, the string version and the closure action. We don't want to test only *one* of those options based on env.

I'd expect something like:

```js
if (!emberVersionGte('3.0')) { // is 3.0 right for this removal?
  test('test sendAction', /* */);
}
if (emberVersionGte('1.13')) {
  test('test (action', /* */);
}
```

...that way the 2.x range of Ember versions is testing both versions, and we don't need the runtime compiler and interpolation.

Reply:
so the only way to do to your recommendation without runtime compiler. would be to duplicate the `fullTable` string, since `onClick="onRowClick"` needs to become `onClick=(action "onRowClick")` or whichever variant. duplicating has a maintenance overhead that I wanted to avoid.

any idea how we can overcome that challenge?

Reply:
@billybonks I think that is the point- we need to have test coverage of both invocation styles. For example you could imagine you even want to test `onClick=this.actionViaActionDecorator` works. That could only be tested in 3.14(ish, whenever the decorator was added) though.

We could write our own babel plugin which converts based on the current version of Ember at build time:

```js
hbsWithClassicActionTranspilation`{{foo onClick=(action 'onRowClick')}}`

// can, based on a macro/transpilation step, be emitted as either:
hbs`{{foo onClick=(action 'onRowClick')}}`
hbs`{{foo onClick='onRowClick'}}`

// in fact in 3.14+ when the @action decorator is used you could also emit it as
hbs`{{foo onClick=this.onRowClick}}`
// thought I guess that requires a different declaration of the action in JS
// space, so maybe is out of scope for a compilation step.
```